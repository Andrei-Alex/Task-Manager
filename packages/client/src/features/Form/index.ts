export * from "./types";
export * from "./data";
export { default as styles } from "./Form.module.scss";
