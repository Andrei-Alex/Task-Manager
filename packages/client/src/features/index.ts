export { default as Form } from "./Form/Form";
export { default as ThemeSwitcher } from "./ThemeSwitcher/ThemeSwitcher";
