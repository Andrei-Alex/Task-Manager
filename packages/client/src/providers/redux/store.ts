import { configureStore } from "@reduxjs/toolkit";
import { AuthSlice } from "./AuthSlice";
import { ThemeSlice } from "./ThemeSlice";

const store = configureStore({
  reducer: {
    auth: AuthSlice.reducer,
    theme: ThemeSlice.reducer,
  },
});
export default store;
