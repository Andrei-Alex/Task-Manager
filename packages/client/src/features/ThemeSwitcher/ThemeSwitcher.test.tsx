import renderer from "react-test-renderer";
import React from "react";
import { ThemeSwitcher } from "@/features";
import store from "../../providers/redux/store";
import { Provider } from "react-redux";

describe("ThemeSwitcher", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeSwitcher />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
