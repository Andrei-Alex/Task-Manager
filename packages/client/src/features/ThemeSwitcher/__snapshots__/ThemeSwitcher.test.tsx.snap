// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeSwitcher renders correctly 1`] = `
<div
  className="mantine-Group-root mantine-k3ov3c"
>
  <div
    className="mantine-Switch-root mantine-Switch-root mantine-1knyxmk"
  >
    <div
      className="mantine-hqx0d1 mantine-Switch-body"
    >
      <input
        checked={false}
        className="mantine-19htmtn mantine-Switch-input"
        id="mantine-4q131l6z4"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="mantine-x21r6k mantine-Switch-track"
        htmlFor="mantine-4q131l6z4"
      >
        <div
          className="mantine-x4nrms mantine-Switch-thumb"
        />
        <div
          className="mantine-jjud08 mantine-Switch-trackLabel"
        >
          <svg
            className="tabler-icon tabler-icon-moon-stars"
            fill="none"
            height="1.25rem"
            stroke="black"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1.5}
            viewBox="0 0 24 24"
            width="1.25rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
            />
            <path
              d="M17 4a2 2 0 0 0 2 2a2 2 0 0 0 -2 2a2 2 0 0 0 -2 -2a2 2 0 0 0 2 -2"
            />
            <path
              d="M19 11h2m-1 -1v2"
            />
          </svg>
        </div>
      </label>
      <div
        className="mantine-xynbgb mantine-Switch-labelWrapper"
      />
    </div>
  </div>
</div>
`;
